pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -112,16 +113,21 @@ func (a *Actuator) isNodeJoin(scope *actuators.MachineScope, controlPlaneMachine
 
 			ec2svc := ec2.NewService(m.Scope)
 
-			ok, err := ec2svc.MachineExists(m)
+			controlplaneExists, err = ec2svc.MachineExists(m)
 			if err != nil {
-				return false, errors.Wrapf(err, "failed to verify existence of machine %q", m.Name())
+				a.log.V(2).Info("Could not verify existence of control plane machine, continuing", "machine-name", scope.Machine.Name, "machine-namespace", scope.Machine.Namespace)
+				continue
 			}
 
-			a.log.V(2).Info("Machine joining control plane", "machine-name", scope.Machine.Name, "machine-namespace", scope.Machine.Name, "should-join-control-plane", ok)
-			return ok, nil
+			if !controlplaneExists {
```

Comment:
you should return true if `controlplaneExists == true`, and return `false` at the end of loop.

currently, it won't work if there are two controlplanes and the first one exists, `controlplaneExists` be false at the end of loop

Reply:
fixed.